tests/utils/crd/crd.go

```diff
@@ -59,3 +62,18 @@ func DefineCustomResourceDefinition(names apiextv1.CustomResourceDefinitionNames
 		},
 	}
 }
+
+func EnsureCrdExists(name string) (bool, error) {
+	apiextv1, err := globalhelper.APIClient.ApiextensionsV1Interface.CustomResourceDefinitions().Get(
+		context.Background(), name, metav1.GetOptions{})
+
+	if err != nil {
```

Comment:
Have you tested this? I think it won't work, as Get() will return an error if the CRD does not exist, and your TC won't be skipped. I think this and some other helper functions can be improved when they use the "Get()" function to get a single resource. You need to use the errors package from the k8s apimachinery:
```
k8sErrors "k8s.io/apimachinery/pkg/api/errors"

func EnsureXXXResourceExists(name string) (bool, error) {
  _, err := globalhelper.APIClient.XXX.Get(...)
  if err != nil {
      if k8sErrors.IsNotFound(err) {
          return false, nil
      } else {
          return false, err
      }
  }
  
  return true, nil 
}
```
If the resource you're asking for does not exist, the Get() function will return a IsNotFound error. It's a bit weird, but as per your function, that's not an error but an expected output when the resource does not exist, so it makes more sense to me to return `false, nil` in that case. If the error is different, then it might be because any other unexpected scenario (faulty connection, permissions, wrong kubeconfig file...). In that case, this unexpected error should be propagated to the caller with `return false, err`.

Reply:
thanks! agreed and fixed.